src/makeInfinityAdder.js

```diff
@@ -7,6 +7,24 @@
 
 function makeInfinityAdder() {
   // write code here
-}
+  let sum = 0;
+  let currentArgs = [];
+
+  function adder(...args) {
+    if (args.length === 0) {
+      const result = sum + currentArgs.reduce((acc, val) => acc + val, 0);
```

Comment:
```suggestion
  let sum = 0;

  function adder(number) {
    if (number === undefined) {
      const result = sum;
```
currentArgs is actually redundant. You can just use recursion with one parameter instead